frontend/test/resolution/testCallInitDeinit.cpp

```diff
@@ -1540,6 +1540,75 @@ static void test19() {
       } );
 }
 
+// Returning a non-nilable 'new' local variable
+static void test20a() {
+  testActions("test20a",
+    R""""(
+      module M {
+        class C {}
+        proc C.init() {}
+        operator C.=(ref lhs: C, rhs: C) { }
+        proc C.deinit() { }
+
+        proc foo() {
+          var x : owned C = new C();
+          return x;
+        }
+      }
+    )"""",
+    {
+      {AssociatedAction::NEW_INIT, "M.foo@5",          ""},
+      {AssociatedAction::DEINIT,   "M.foo@9",          "x"},
+    });
+}
+
+// Generic instantiation version
+/* static void test20b() { */
+/*   testActions("test20b", */
+/*     R""""( */
+/*       module M { */
+/*         class C { */
+/*           var x; */
+/*         } */
+/*         proc C.init(x) {this.x = x;} */
+/*         operator C.=(ref lhs: C(?), rhs: C(?)) { } */
+/*         proc C.deinit() { } */
+
+/*         proc foo() { */
+/*           var x : owned C = new C(3); */
+/*           return x; */
+/*         } */
+/*       } */
+/*     )"""", */
+/*     { */
+/*       {AssociatedAction::NEW_INIT,   "M.foo@6",    ""}, */
+/*       {AssociatedAction::INIT_OTHER, "x",          ""}, */
+/*       {AssociatedAction::DEINIT,     "M.foo@10",   "x"}, */
+/*     }); */
+/* } */
```

Comment:
If this is meant as a future TODO once this behavior comes online, can you add a comment?

Reply:
Future TODO, adding a comment.